subprojects/docs/src/docs/userguide/migrating_from_ant.adoc

```diff
@@ -0,0 +1,328 @@
+
+Retrieval::
+As mentioned, Gradle does not automatically copy files from the dependency cache. Its standard tasks typically use the files directly. If you want to copy the dependencies to a local directory, you can use something like this in your build script:
++
+[source,groovy]
+----
+task retrieveRuntimeDependencies(type: Copy) {
+    into 'libs'
+    from configurations.runtimeClasspath
+}
+----
++
+A configuration is also a file collection, hence why it can be used in the `from()` configuration. You can use a similar technique to attach a configuration to a compilation task or one that produces documentation. See the chapter on <<working_with_files#working_with_files,Working with Files>> for more examples and information on Gradle's file API.
+
+== Publishing artifacts
+
+Projects that use Ivy to manage dependencies often use it for publishing JARs and other artifacts to repositories. If you're migrating such a build, then you'll be glad to know that Gradle has built-in support for publishing artifacts to Ivy-compatible repositories.
+
+Before you attempt to migrate this particular aspect of your build, read the <<publishing_overview#publishing_overview,publishing overview>> chapter to learn about Gradle's publishing model. That chapter's examples are based on Maven repositories, but the same model is used for Ivy repositories as well.
+
+The basic migration process looks like this:
+
+ * Apply the <<publishing_ivy#publishing_ivy,Ivy Publish Plugin>> to your build
+ * <<publishing_ivy#publishing_ivy:publications,Configure at least one publication>>, representing what will be published (including additional artifacts if desired)
+ * <<publishing_ivy#publishing_ivy:repositories,Configure one or more repositories to publish artifacts to>>
+
+Once that's all done, you'll be able to generate an Ivy module descriptor for each publication and publish them to one or more repositories.
+
+Let's say you have defined a publication named "myLibrary" and a repository named "myRepo". The Ivy tasks would then map to the Gradle tasks like this:
+
+ * `<deliver>` -> `generateDescriptorFileForMyLibraryPublication`
+ * `<publish>` -> `publishMyLibraryPublicationToMyRepoRepository`
+
+There is also a `publish` task that publishes all publications to all repositories. You could also define your own aggregate task that depends on just a subset of those publishing tasks.
+
+.On dependency versions
+[NOTE]
+====
+Ivy will, by default, automatically replace dynamic versions of dependencies with the resolved "static" versions when it generates the module descriptor. Gradle does _not_ mimic this behavior: declared dependency versions are left unchanged.
```

Comment:
Would it be appropriate to mention Gradle's dependency locking here?

Reply:
Possibly. May be worth mentioning it somewhere anyway.